### ISSUE TITLE
Ghost results in character assets

### ISSUE THREAD

**Issue by guiguilechat** [esi/esi-issues#792]
I have a toon which has

 - one fitted empty epithal 
 - no contract 
 - nothing in safety
 - nothing else in the ingame assets browser

Yet I have assets that appear in the assets endpoint.

The return has only one page ( x-pages: 1  )
I removed the correct results, here is what remains : 

```
[
  {
    "type_id": 34,
    "quantity": 1,
    "location_id": 1026315194566,
    "location_type": "other",
    "item_id": 1026315194570,
    "location_flag": "Cargo",
    "is_singleton": false
  },
  {
    "type_id": 8529,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437039989,
    "location_flag": "MedSlot1",
    "is_singleton": true
  },
  {
    "type_id": 21857,
    "quantity": 1,
    "location_id": 1025485814277,
    "location_type": "other",
    "item_id": 1025485814280,
    "location_flag": "MedSlot0",
    "is_singleton": true
  },
  {
    "type_id": 21857,
    "quantity": 1,
    "location_id": 1026315194566,
    "location_type": "other",
    "item_id": 1026315194569,
    "location_flag": "MedSlot0",
    "is_singleton": true
  },
  {
    "type_id": 35659,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437023732,
    "location_flag": "MedSlot0",
    "is_singleton": true
  },
  {
    "type_id": 3638,
    "quantity": 1,
    "location_id": 1025485814277,
    "location_type": "other",
    "item_id": 1025485814278,
    "location_flag": "HiSlot0",
    "is_singleton": true
  },
  {
    "type_id": 3638,
    "quantity": 1,
    "location_id": 1026315194566,
    "location_type": "other",
    "item_id": 1026315194567,
    "location_flag": "HiSlot0",
    "is_singleton": true
  },
  {
    "type_id": 3651,
    "quantity": 1,
    "location_id": 1025485814277,
    "location_type": "other",
    "item_id": 1025485814279,
    "location_flag": "HiSlot1",
    "is_singleton": true
  },
  {
    "type_id": 3651,
    "quantity": 1,
    "location_id": 1026315194566,
    "location_type": "other",
    "item_id": 1026315194568,
    "location_flag": "HiSlot1",
    "is_singleton": true
  },
  {
    "type_id": 2537,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024879453165,
    "location_flag": "MedSlot2",
    "is_singleton": true
  },
  {
    "type_id": 2537,
    "quantity": 1,
    "location_id": 1026315254995,
    "location_type": "other",
    "item_id": 1026315254974,
    "location_flag": "MedSlot3",
    "is_singleton": true
  },
  {
    "type_id": 9580,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024879451495,
    "location_flag": "MedSlot3",
    "is_singleton": true
  },
  {
    "type_id": 10998,
    "quantity": 1,
    "location_id": 1026315254995,
    "location_type": "other",
    "item_id": 1026315255014,
    "location_flag": "LoSlot1",
    "is_singleton": true
  },
  {
    "type_id": 10998,
    "quantity": 1,
    "location_id": 1026315254995,
    "location_type": "other",
    "item_id": 1026315258410,
    "location_flag": "LoSlot0",
    "is_singleton": true
  },
  {
    "type_id": 11577,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437026394,
    "location_flag": "HiSlot0",
    "is_singleton": true
  },
  {
    "type_id": 11577,
    "quantity": 1,
    "location_id": 1026315254995,
    "location_type": "other",
    "item_id": 1026315255016,
    "location_flag": "HiSlot0",
    "is_singleton": true
  },
  {
    "type_id": 1405,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437012987,
    "location_flag": "LoSlot4",
    "is_singleton": true
  },
  {
    "type_id": 1405,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437017738,
    "location_flag": "LoSlot0",
    "is_singleton": true
  },
  {
    "type_id": 1405,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437018250,
    "location_flag": "LoSlot2",
    "is_singleton": true
  },
  {
    "type_id": 2605,
    "quantity": 1,
    "location_id": 1026315254995,
    "location_type": "other",
    "item_id": 1026315254987,
    "location_flag": "LoSlot2",
    "is_singleton": true
  },
  {
    "type_id": 1319,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437016422,
    "location_flag": "LoSlot3",
    "is_singleton": true
  },
  {
    "type_id": 1319,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024437017742,
    "location_flag": "LoSlot1",
    "is_singleton": true
  },
  {
    "type_id": 31119,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024879447644,
    "location_flag": "RigSlot2",
    "is_singleton": true
  },
  {
    "type_id": 31119,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024879447960,
    "location_flag": "RigSlot0",
    "is_singleton": true
  },
  {
    "type_id": 31119,
    "quantity": 1,
    "location_id": 1024354560327,
    "location_type": "other",
    "item_id": 1024879448063,
    "location_flag": "RigSlot1",
    "is_singleton": true
  }
]
```

I try to resolve the location_ids : 
[GET]https://esi.tech.ccp.is/latest/universe/structures/1024354560327/ data=null 403{"error":"Forbidden"}
[GET]https://esi.tech.ccp.is/latest/universe/structures/1025485814277/ data=null 403{"error":"Forbidden"}
[GET]https://esi.tech.ccp.is/latest/universe/structures/1026315194566/ data=null 403{"error":"Forbidden"

It looks like the **566 location is a noobship with 1 tritanium and standard noobship fit.
However that noobship is not shown anywhere in my assets ingame.
My epithal shows in the result, as an item_id of type_id 655 and placed in a correct location_id ; the fit is shown as several item_id of correct type_id and placed in the location_id corresponding to the epithal's item_id ; This is correct for the epithal.

Basically I am missing an entry consisting of the item_id 1026315194566 , with correct type_id and location_id associated.
My guess is that some orphan data remain in the database when the ship is deleted(I guess I destroyed it)

**Comment by tsuthers-ccpgames** [comment]
> My guess is that some orphan data remain in the database when the ship is deleted(I guess I destroyed it)

Spot on, we're not recursing far enough up the tree to see when something is in the junkyard if it's not at the top level.

**Comment by ccp-zoetrope** [comment]
This is out now, would like a confirmation that it's working to consider it closed :)

**Comment by hedge00** [comment]
Holy crow that's a big QOL improvement. As a prolific highsec ganker and a jEveAssets user, my assets were constantly cluttered up by thousands of items and ships I had expended, to the tune of 2 billion worth of destroyers and noobships that didn't really exist. I did an update after your fix went live and those are ALL GONE now. Thanks a lot for the fix!

**Comment by jowrjowr** [comment]
looks resolved. closed.

**Comment by guiguilechat** [comment]
I don't have time to check this, if other people say it works,  I guess it's corrected.
